src/librustc/middle/expr_use_visitor.rs

```diff
@@ -276,15 +276,13 @@ enum PassArgs {
 }
 
 impl<'d,'t,'a,'tcx> ExprUseVisitor<'d,'t,'a,'tcx> {
-    pub fn new(delegate: &'d mut Delegate<'tcx>,
+    pub fn new(delegate: &'d mut (Delegate<'tcx>),
                typer: &'t infer::InferCtxt<'a, 'tcx>)
                -> ExprUseVisitor<'d,'t,'a,'tcx>
     {
-        ExprUseVisitor {
-            typer: typer,
-            mc: mc::MemCategorizationContext::new(typer),
-            delegate: delegate,
-        }
+        let mc: mc::MemCategorizationContext<'t, 'a, 'tcx> =
+            mc::MemCategorizationContext::new(typer);
+        ExprUseVisitor { typer: typer, mc: mc, delegate: delegate }
```

Comment:
@eddyb 

>  I've had issues with this myself, do you know what's up here?

As far as I know, it is the bug fixed by https://github.com/rust-lang/rust/pull/27892, but not yet snapshotted. However, it could be that we did a snapshot, in which case it is because we need to rework region inference. I've been putting that off but maybe the time has come.